outputs/policyreport.go

```diff
@@ -0,0 +1,325 @@
+package outputs
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/falcosecurity/falcosidekick/types"
+	"github.com/google/uuid"
+	"github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/kube-bench-adapter/pkg/apis/wgpolicyk8s.io/v1alpha2"
+	wgpolicy "github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/kube-bench-adapter/pkg/apis/wgpolicyk8s.io/v1alpha2"
+	crdClient "github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/kube-bench-adapter/pkg/generated/v1alpha2/clientset/versioned"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	clusterPolicyReportBaseName = "falco-cluster-policy-report-"
+	policyReportBaseName        = "falco-policy-report-"
+	policyReportSource          = "Falco"
+	highpriority                = "high"
+	lowpriority                 = "low"
+	mediumpriority              = "medium"
+	fail                        = "fail"
+	warn                        = "warn"
+)
+
+var (
+	minimumPriority string
+	severity        string
+	result          string
+	//slice of policy reports
+	policyReports = make(map[string]*wgpolicy.PolicyReport)
+	//cluster policy report
+	clusterPolicyReport *wgpolicy.ClusterPolicyReport = &wgpolicy.ClusterPolicyReport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterPolicyReportBaseName,
+		},
+		Summary: v1alpha2.PolicyReportSummary{
+			Fail: 0,
+			Warn: 0,
+		},
+	}
+)
+
+func NewPolicyReportClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
+	clusterPolicyReport.ObjectMeta.Name += uuid.NewString()[:8]
+	minimumPriority = config.PolicyReport.MinimumPriority
+
+	clientConfig, err := rest.InClusterConfig()
```

Comment:
I don't understand that block, why are you setting `clientConfig` twice, with one time in a condition. Have you tried to connect to k8s api inside and outside a pod?

Reply:
yes, it's being done to check the condition if it's being executed inside the cluster or outside the cluster, let me know if we need only for the outside cluster, we can change it accordingly then.
This seemed appropriate back when I was writing the code.

Reply:
it's ok, we must have access from inside cluster too